api/channel.go

```diff
@@ -397,10 +397,15 @@ func updateChannelPurpose(c *Context, w http.ResponseWriter, r *http.Request) {
 }
 
 func getChannels(c *Context, w http.ResponseWriter, r *http.Request) {
+	// check if the team exist
+	if result := <-Srv.Store.Team().Get(c.TeamId); result.Err != nil {
```

Comment:
The issue is when you also send a wrong teamid and it return all the DM channels.
Scenario
you have some DM channel and request the getchannel sending a wrong teamid this will return the dm channels.

My understanding here is check the teamid if it is valid (exist) then continue otherwise send a 400 http status.